### DISCUSSION TITLE
Any trick for apps running - to restart after update (Alfred, nextcloud, ...cloud)

### DISCUSSION THREAD

**Discussion by djheidihoe** [Homebrew/discussions#3104] [Casks]
Hi Community,

i tried the search but couldn't find anything.
So i try it here.

My userstory is the following:
i run a mac and have Alfred4, nextcloud and other running in the background.

after brew update & brew upgrade i get all the latest versions. (i really like updating all my apps from terminal in this easy way)
my (minor) issue is, after the update of i.E. Alfred4 and nextcloud the apps are not running anymore.
This leads me to non working shortcuts und missing file sync. It is more of an annoyance with my mac mini if i do the update from remote ssh.

So is there a way to auto add "open -a Alfred\ 4" which i need to do after every update on several machines?

Thanks in advance

**Comment by toobuntu** [comment]
This is possible, though a bit cumbersome. You can see how I do it in [Babble](https://github.com/toobuntu/babble).

Adding this to Homebrew is [not simple](https://github.com/Homebrew/homebrew-cask/issues/104074).

The crux of this approach is to:

1. Maintain a list of apps known to require a restart after being updated.
2. Generate a list of those which were running prior to the upgrade.
3. Check that list again after the upgrade and restart apps as necessary.

To implement this, I used the apps' bundle IDs instead of their names because they should not contain empty spaces and that was helpful for scripting this. You can get those with something like

```shell
osascript -e 'id of app "Alfred"'
```

, for example.

Then, you can use `/usr/bin/lsappinfo` to find out which of those are currently running, both before and again after the `brew upgrade`. If it is not running afterward, then use something like

```shell
/usr/bin/open -b com.runningwithcrayons.Alfred
```

to open it again.

Hope this helps, in case you are still searching for a solution for this.
